prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala

```diff
@@ -90,6 +92,7 @@ class ParserSpec extends FunSpec with Matchers {
     parseSuccessfully("foo * bar")
     parseSuccessfully("foo == 1")
     parseSuccessfully("foo == bool 1")
+    parseSuccessfully("scalar(foo) > bool scalar(bar)")
```

Comment:
Need a test case that will parse `foo > bool bar` like binary join expressions with an error

Reply:
This query is valid. It will get parsed to BinaryJoinExec with GTR_BOOL. I will add a test case